### PR TITLE
Added an option to prevent some subscriptions

### LINKED ISSUE #286: Ability to avoid some subscriptions to document events.
Is it possible to opt out of some subscriptions in registerHandlers method of ContextMenu. It would be great to have more control over it with props, just to add flexibility in some tricky use cases.

### PR DESCRIPTION
**Pull request by rSensation**
closes #286 

Added props: **preventHideOnContextMenu**, **preventHideOnScroll** and **preventHideOnResize** to give more flexibility to users.

### CODE REVIEW

File: src/ContextMenu.js [hunk 1]
```diff
@@ -89,10 +95,10 @@ export default class ContextMenu extends AbstractMenu {
     registerHandlers = () => {
         document.addEventListener('mousedown', this.handleOutsideClick);
         document.addEventListener('touchstart', this.handleOutsideClick);
-        document.addEventListener('scroll', this.handleHide);
-        document.addEventListener('contextmenu', this.handleHide);
+        if (!this.props.preventHideOnScroll) document.addEventListener('scroll', this.handleHide);
+        if (!this.props.preventHideOnContextMenu) document.addEventListener('contextmenu', this.handleHide);
         document.addEventListener('keydown', this.handleKeyNavigation);
-        window.addEventListener('resize', this.handleHide);
+        if (!this.props.preventHideOnResize) window.addEventListener('resize', this.handleHide);
```
Comment (vkbansal) [inline src/ContextMenu.js]: Probably add the same in `unregisterHandlers`?

Comment (rSensation) [reply src/ContextMenu.js]: I'm worried about a possibility that any of this props will change from false to true after `registerHandlers` but before unregister. It seems safer to unsubscribe without condition. What do you think?

Comment (vkbansal) [reply src/ContextMenu.js]: Yeah that sounds legit

**Comment by vkbansal** [comment]
Thanks for the contribution. There is conflict in `package-lock.json`. Please fix it. I'll merge this after that

**Pull request merged**